### PR TITLE
Small cleanups to file checks and associated documentation

### PR DESCRIPTION
**Pull request by fmahebert**
### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: docs/DevGuide/Travis.md [hunk 1]
```diff
@@ -48,26 +48,30 @@ indicate that a pull request should not be merged in its current state.
...
+      See `tools/FileTestDefs.sh` for the full list of exceptions.)
+    * is missing the license line
+    * does not end with a newline
+    * contains a tab character
+    * contains white space at the end of a line
+    * contains a carriage return character
+  - A `c++` header file (i.e., `*.hpp` or `*.tpp`) is missing `#%pragma once`
+  - A `c++` file (i.e., `*.hpp`, `*.tpp`, or `*.cpp`) file,
+    * includes `<iostream>` (useless when running in parallel)
+    * includes `<lrtslock.h>` (use `converse.h` instead)
```
Comment (nilsdeppe) [inline docs/DevGuide/Travis.md]: [optional] consider `<coverse.h>` (angle brackets)

File: docs/DevGuide/Travis.md [hunk 2]
```diff
@@ -48,26 +48,30 @@ indicate that a pull request should not be merged in its current state.
...
+    * includes `Utilities/TmplDebugging.hpp` (used only for debugging)
```
Comment (nilsdeppe) [inline docs/DevGuide/Travis.md]: [optional] Consider quotes around path

**Review by nilsdeppe — DISMISSED** [review]
LGTM! A couple optional things that you may care about :) Feel free to squash if you want to change, but I'm totally fine either way :)

**Comment by fmahebert** [comment]
Thanks for nitpicking me @nilsdeppe ;) .. I agree with your suggestions so I've squashed them in.

**Review by nilsdeppe — APPROVED** [review]
You're welcome :) Thanks for doing this

**Review by nilsvu — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**